File: Creational/Prototype/Php/Example/Circle.php
<?php

/**
 * Circle implementation of Shape prototype
 */
class Circle extends Shape
{
    private float $radius;
    
    public function __construct(float $radius = 0, string $color = '', int $x = 0, int $y = 0)
    {
        parent::__construct();
        $this->radius = $radius;
        $this->color = $color;
        $this->x = $x;
        $this->y = $y;
        if ($radius > 0) {
            echo "Circle created: radius={$radius}, color={$color}\n";
        }
    }
    
    public function clone(): Prototype
    {
        $clone = new self($this->radius, $this->color, $this->x, $this->y);
        echo "Circle cloned: radius={$this->radius}, color={$this->color}\n";
        return $clone;
    }
    
    public function getInfo(): string
    {
        return sprintf(
            "Circle[radius=%.1f, color='%s', position=(%d,%d), area=%.2f]",
            $this->radius,
            $this->color,
            $this->x,
            $this->y,
            $this->getArea()
        );
    }
    
    public function getArea(): float
    {
        return pi() * $this->radius * $this->radius;
    }
    
    public function getRadius(): float
    {
        return $this->radius;
    }
    
    public function setRadius(float $radius): void
    {
        $this->radius = $radius;
    }
}

File: Creational/Prototype/Java/Example/Circle.java
/**
 * Circle implementation of Shape prototype
 */
public class Circle extends Shape {
    private double radius;
    
    public Circle() {}
    
    public Circle(double radius, String color, int x, int y) {
        this.radius = radius;
        this.color = color;
        this.x = x;
        this.y = y;
        System.out.println("Circle created: radius=" + radius + ", color=" + color);
    }
    
    public Circle(Circle other) {
        super(other);
        if (other != null) {
            this.radius = other.radius;
        }
        System.out.println("Circle cloned: radius=" + radius + ", color=" + color);
    }
    
    @Override
    public Prototype clone() {
        return new Circle(this);
    }
    
    @Override
    public String getInfo() {
        return String.format("Circle[radius=%.1f, color='%s', position=(%d,%d), area=%.2f]", 
                           radius, color, x, y, getArea());
    }
    
    @Override
    public double getArea() {
        return Math.PI * radius * radius;
    }
    
    public double getRadius() {
        return radius;
    }
    
    public void setRadius(double radius) {
        this.radius = radius;
    }
}